app/models/email_access_token.rb

```diff
@@ -15,8 +17,8 @@ class EmailAccessToken < ApplicationRecord
   validates_presence_of :email_address
   validate :one_or_more_valid_email_addresses
 
-  scope :by_raw_token, ->(raw_token) do
-    where(token: Devise.token_generator.digest(EmailAccessToken, :token, raw_token))
+  scope :lookup, ->(raw_token) do
```

Comment:
I thought that there was a "devise way" to create auto-expiring tokens, but this is also super valid and fine!

Reply:
I was curious how Devise/devise-invitations does this, so I just checked. `devise-invitations` exposes a config option called `invite_for` which powers a method in that Gem called `valid_invitation?` (which we call via the Gem's `find_by_invitation_token` method from our InvitationsController's `require_valid_invitation_token`).